actionview/lib/action_view/base.rb

```diff
@@ -237,11 +237,23 @@ def initialize(lookup_context, assigns, controller) # :nodoc:
       _prepare_context
     end
 
-    def _run(method, template, locals, buffer, add_to_stack: true, &block)
+    def _run(method, template, locals, buffer, add_to_stack: true, locals_disabled: false, explicit_locals: false, &block)
       _old_output_buffer, _old_virtual_path, _old_template = @output_buffer, @virtual_path, @current_template
       @current_template = template if add_to_stack
       @output_buffer = buffer
-      public_send(method, locals, buffer, &block)
+
+      if !explicit_locals
```

Comment:
I specifically tried to put the existing branch logic first for performance reasons. I'm partial towards `if !` vs. `unless`. (same for the conditional below)

Reply:
I really don't think this is going to make any noticeable performance difference so I'd flip the conditions for readability.